app/services/alerts/alert_attributes_factory.rb

```diff
@@ -1,18 +1,19 @@
 module Alerts
   class AlertAttributesFactory
-    def initialize(school, alert_report, alert_generation_run, alert_type)
+    def initialize(school, alert_report, alert_generation_run, alert_type, asof_date)
       @school = school
       @alert_report = alert_report
       @alert_generation_run = alert_generation_run
       @alert_type = alert_type
+      @asof_date = asof_date
     end
 
     def generate
       {
         school_id:                @school.id,
         alert_generation_run_id:  @alert_generation_run.id,
         alert_type_id:            @alert_type.id,
-        run_on:                   @alert_report.asof_date,
+        run_on:                   @asof_date,
```

Comment:
This is never set in the alert report (was at one point) therefore pass it in to set the run on date.